### ISSUE TITLE
NServiceBus Unbounded Integration Test fails with          Exception occurred in try number 2 : System.Net.WebException: Unable to connect to the remote server ---> System.Net.Sockets.SocketException: No connection could be made because the target machine actively refused it

### ISSUE THREAD

**Issue by jaffinito** [newrelic/newrelic-dotnet-agent#361]
**Description**
The NServiceBus unbounded integration test is always failing with the following:

```
Exception occurred in try number 2 : System.Net.WebException: Unable to connect to the remote server ---> System.Net.Sockets.SocketException: No connection could be made because the target machine actively refused it 172.21.80.1:52185
           at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
           at System.Net.ServicePoint.ConnectSocketInternal(Boolean connectFailure, Socket s4, Socket s6, Socket& socket, IPAddress& address, ConnectSocketState state, IAsyncResult asyncResult, Exception& exception)
           --- End of inner exception stack trace ---
           at System.Net.WebClient.DownloadDataInternal(Uri address, WebRequest& request)
           at System.Net.WebClient.DownloadString(Uri address)
           at NewRelic.Agent.UnboundedIntegrationTests.RemoteServiceFixtures.NServiceBusBasicMvcApplicationFixture.GetMessageQueue_NServiceBus_Send() in D:\a\newrelic-dotnet-agent\newrelic-dotnet-agent\tests\Agent\IntegrationTests\UnboundedIntegrationTests\RemoteServiceFixtures\NServiceBusBasicMvcApplicationFixture.cs:line 26
           at NewRelic.Agent.IntegrationTestHelpers.RemoteServiceFixtures.RemoteApplicationFixture.Initialize() in D:\a\newrelic-dotnet-agent\newrelic-dotnet-agent\tests\Agent\IntegrationTests\IntegrationTestHelpers\RemoteServiceFixtures\RemoteApplicationFixture.cs:line 239
```

This example is from: https://github.com/newrelic/newrelic-dotnet-agent/runs/1387255064?check_suite_focus=true#step:9:298

This unbounded test is one of only a few that run directly on the GH runner alongside the test itself (MSQM is another).  There is not remote server, just the local server.

**Comment by nr-ahemsath** [comment]
I fixed an issue with this test which was causing it to fail locally, but it is still failing in GHA.  Reopening.

**Comment by nr-ahemsath** [comment]
I made another effort to fix this.  The proximate cause of this test's failures are the inability to start a web host with HostedWebCore for some unknown reason.  This is often the cause of other tests' intermittent failures as well.

Recently we updated another group of queue instrumentation tests for `RabbitMQ` to use the ConolseMultiFunction{FW;Core}Application mechanism, since there's nothing inherent in this test that requires the library methods being exercised for the purpose of verifying that we instrument them be hosted in a web application.  I attempted to do the same with the NServiceBus instrumentation tests.  However, I was unable to get a working console application that would send and receive an NServiceBus message to and within itself.  I followed an NServiceBus tutorial for sending messages and registering a message handler for that message's type, but it is not working.  The sending part appears to work as expected, but the registered message handler never gets called.  Here's where the sample project is stashed for future reference: https://source.datanerd.us/ahemsath/sandbox/tree/master/NServiceBusDemo

I was more or less following this tutorial: https://docs.particular.net/tutorials/nservicebus-step-by-step/1-getting-started/  However, this tutorial appears to be written for a newer version of NServiceBus than the 5.x we currently support and need to test.  The main salient difference I'm aware of at this point is that the `LearningTransport` type doesn't exist in 5.x, so that might have something to do why what I'm doing isn't working.
